cmd/treedb_vector_partition_bench/main.go

```
@@ -0,0 +1,1494 @@
+			result, err := simulate(cfg, fixture, vectors, queries, probes, overlap)
+			if err != nil {
+				return err
+			}
+			if err := validateResult(result); err != nil {
+				return err
+			}
+			if err := writeArtifacts(cfg.out, result); err != nil {
+				return err
+			}
+			if cfg.format == "json" {
+				b, _ := json.Marshal(result)
```

golangci-lint (2.12.2)

[error] Error return value of `encoding/json.Marshal` is not checked: unsafe type `float64` found

(errchkjson)